abdiff/core/utils.py

```diff
@@ -78,6 +79,29 @@ def load_dataset(base_dir: str | Path) -> ds.Dataset:
     return ds.dataset(base_dir, partitioning="hive")
 
 
+def parse_timdex_filename(s3_uri_or_filename: str) -> dict[str, str | None]:
+    """Parse details from filename."""
+    filename = s3_uri_or_filename.split("/")[-1]
+
+    match_result = re.match(
+        r"^([\w\-]+?)-(\d{4}-\d{2}-\d{2})-(\w+)-(\w+)-records-to-(.+?)(?:_(\d+))?\.(\w+)$",
+        filename,
+    )
+
+    keys = ["source", "date", "cadence", "stage", "action", "index", "file_type"]
```

Comment:
Apologies!  Missed this on first pass.

Can we update these now, as discussed we might do in the TIMDEX parquet work? 

- `date` --> `run-date`
- `cadence` --> `run-type`

Reply:
Here is a comment from the engineering plan: https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/edit-v2/4094296066?focusedCommentId=4141219863.

The name change is not performed yet in this engineering plan, but I think it was a good idea and will make that change.

Reply:
Good catch! Made the required updates. See latest commit [9425ce3](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/42/commits/9425ce3b4a3c44267944687387623b73bfc3af67). 🤓